examples/thermostat/thermostat-common/src/atomic-write-manager.cpp

```diff
@@ -0,0 +1,535 @@
+{
+    // These are the only two atomic attributes
+    if (attributeId.has_value() && attributeId.value() != Presets::Id && attributeId.value() != Schedules::Id)
+    {
+        return false;
+    }
+
+    bool inAtomicWrite = false;
+    uint16_t ep =
+        emberAfGetClusterServerEndpointIndex(endpoint, Thermostat::Id, MATTER_DM_THERMOSTAT_CLUSTER_SERVER_ENDPOINT_COUNT);
+
+    if (ep < ArraySize(mAtomicWriteSessions))
+    {
+        inAtomicWrite = mAtomicWriteSessions[ep].state == kAtomicWriteState_Open;
+    }
+    return inAtomicWrite;
+}
+
+bool ThermostatAtomicWriteManager::InWrite(const std::optional<AttributeId> attributeId,
+                                           const Access::SubjectDescriptor & subjectDescriptor, EndpointId endpoint)
+{
+    if (!InWrite(attributeId, endpoint))
+    {
+        return false;
+    }
+    return subjectDescriptor.authMode == Access::AuthMode::kCase &&
+        GetAtomicWriteScopedNodeId(attributeId, endpoint) == ScopedNodeId(subjectDescriptor.subject, subjectDescriptor.fabricIndex);
+}
+
+bool ThermostatAtomicWriteManager::InWrite(const std::optional<AttributeId> attributeId, CommandHandler * commandObj,
+                                           EndpointId endpoint)
+{
+    if (!InWrite(attributeId, endpoint))
+    {
+        return false;
+    }
+    ScopedNodeId sourceNodeId = GetSourceScopedNodeId(commandObj);
+    return GetAtomicWriteScopedNodeId(attributeId, endpoint) == sourceNodeId;
+}
+
+bool ThermostatAtomicWriteManager::InWrite(DataModel::DecodableList<AttributeId>::Iterator attributeIds,
+                                           CommandHandler * commandObj, EndpointId endpoint)
+{
+    while (attributeIds.Next())
+    {
+        auto & attributeId = attributeIds.GetValue();
+        if (!InWrite(attributeId, commandObj, endpoint))
+        {
+            return false;
+        }
+    }
+    if (attributeIds.GetStatus() != CHIP_NO_ERROR)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool ThermostatAtomicWriteManager::BeginWrite(chip::app::CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                              const Commands::AtomicRequest::DecodableType & commandData)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    size_t attributeRequestCount = 0;
+
+    if (mDelegate == nullptr)
+    {
+        commandObj->AddStatus(commandPath, imcode::InvalidInState);
+        return false;
+    }
+
+    auto status = countAttributeRequests(commandData, attributeRequestCount);
+    if (status != imcode::Success)
+    {
+        commandObj->AddStatus(commandPath, status);
+        return false;
+    }
+    if (attributeRequestCount == 0)
+    {
+        commandObj->AddStatus(commandPath, imcode::InvalidCommand);
+        return false;
+    }
+
+    auto timeoutRequest = System::Clock::Milliseconds16(commandData.timeout.Value());
```

Comment:
this seems to be used only once. maybe just use System::Clock::Milliseconds16(commandData.timeout.Value()) at line [here](https://github.com/project-chip/connectedhomeip/pull/34936/files#diff-1e54d0c08c51e7226f19f1461cb899ae549d766577aed32b612873b7f6104cf1R375) or move this line down to narrow the scope.

Reply:
Will move down.